tests/rustworkx_tests/digraph/test_transitive_reduction.py

```diff
@@ -0,0 +1,73 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import rustworkx
+
+
+class TestTransitiveReduction(unittest.TestCase):
+    def test_tr(self):
+        graph = rustworkx.PyDiGraph()
+        a = graph.add_node("a")
+        b = graph.add_node("b")
+        c = graph.add_node("c")
+        d = graph.add_node("d")
+        e = graph.add_node("e")
+        graph.add_edges_from(
+            [(a, b, 1), (a, d, 1), (a, c, 1), (a, e, 1), (b, d, 1), (c, d, 1), (c, e, 1), (d, e, 1)]
+        )
+        tr, _ = rustworkx.transitive_reduction(graph)
+        self.assertCountEqual(list(tr.edge_list()), [(0, 2), (0, 1), (1, 3), (2, 3), (3, 4)])
```

Comment:
Doesn't this only assert that the number of edges is matching the expectation. Shouldn't this assert the edge lists match the expected edges in the reduction?

Reply:
I think `assertCountEqual()` asserts that the list has the same elements, disregarding order. Here are the [docs](https://docs.python.org/3.2/library/unittest.html#unittest.TestCase.assertCountEqual) in case I'm reading it wrong. I'm just trying to see if the lists have the same elements and checking `assertListEqual()` fails because the order differs

Reply:
Ah, yeah you're correct. I was misremembering what the method does. This does check the element counts are the same in the two lists which is what we want here.